### PR TITLE
improved spellcheck

### PR DESCRIPTION
**Pull request by chirayu-humar**
Closes - https://github.com/plone/documentation/issues/1190
Fixex "#`1190`"
* changed to "NodeJS" and "JavaScript".
* added "entry of change log file" with name "1190.internal".
* added screen shots of changed code.

accept
reject
entry_to_change_log_file

### CODE REVIEW

**Comment by chirayu-humar** [comment]
Hello @stevepiercy,

I hope you're doing well! I have prepared a draft pull request (`1190`) that improves spellcheck. Since you are the maintainer familiar with this area, I would greatly appreciate your expert feedback and review.

This pull request aims to address issue `1190` and adds functionality to improve the user experience.

Please let me know if you have time to take a look at it whenever you get a chance. I'm open to any suggestions, and I'm eager to make any necessary adjustments to align with the project's guidelines and standards.

Thank you in advance for your time and consideration.

Best regards,
@chirayu-humar ,

**Comment by stevepiercy** [comment]
@chirayu-humar sorry, this PR is the exact opposite of what we need. See:

https://6.docs.plone.org/contributing/documentation/setup-build.html#vale

> Plone documentation uses a custom spelling dictionary, with accepted and rejected spellings in styles/Vocab/Plone.

Words in `reject.txt` are rejected, whereas those in `accept.txt` are accepted. The existing entries in `reject.txt` should be rejected. Please do not remove them.

**Comment by chirayu-humar** [comment]
ok, i feel now i understood, i have to include "JavaScript", "NodeJS" in accept.txt and "Javascript", "Js", "node.js" and "node" etc in reject.txt, am i write ?

**Comment by stevepiercy** [comment]
Let's take a step back. Vale already has a built-in dictionary against which it checks spelling of thousands of words.

Vale will reject some words that we want to accept, such as `Volto`. We put such words in `accept.txt`.

Similarly Vale might accept words that we want to reject. This includes improper casing, such as "Javascript", and incorrectly spelled words. We put such words in `reject.txt`.

That is how we customize spell checking.

To run spell checking, you can call `make vale`.

You can also call Vale directly with specific commands to improve the spelling of a single file or directory of files. `vale -h` will display Vale's help.

**Comment by stevepiercy** [comment]
I'm sorry, I don't know how to explain further. The correct spelling of "JavaScript" is exactly that, not "javascript", "Javascript", or "js". That's why the [first is in `accept.txt`](https://github.com/plone/volto/blob/master/styles/Vocab/Plone/accept.txt#L9) and the [other three are in `reject.txt`](https://github.com/plone/volto/blob/master/styles/Vocab/Plone/reject.txt#L3-L5).

These serve as examples of how to find other words that should be accepted or rejected throughout the documentation and added to those two files.

When you run `make vale`, there should be a lot of spelling warnings that could benefit from proper spelling and casing configuration.

I don't think you can use regular expressions in these two files, but you can use [ignore files and filters](https://vale.sh/docs/topics/styles/#spelling) in the configuration.

**Comment by chirayu-humar** [comment]
have installed vale and tried running command "make vale" but there is no such rule named "vale" but there is one which i identified, which is named "docs-vale".

**Comment by stevepiercy** [comment]
Ah, yes, for non-documentation repos, we use the prefix `docs-` for `make` commands. Sorry about that. Please use `make docs-vale` instead, and let me know.

File: .github/workflows/docs.yml
Comment (stevepiercy) [inline .github/workflows/docs.yml:41]: Revert.

File: styles/Vocab/Plone/reject.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/reject.txt:8]: Remove. Duplicate of line 2.

```suggestion
```

**Comment by chirayu-humar** [comment]
hi @stevepiercy, 
fortunately at last command "make docs-vale" ran successfully, on the bassis of suggestions it gave to me in output i have commited a chage to pull request, added "plone", "volto" and "razzle" to reject.txt file, because these 3 are wrong casing configurations and there write once are already written in accept.js so they are not needed to be added again, 

but now, i just request you sir, please review my pull request and guide whether my approach is write ?, i beleave that i have to put all the words after "instead" into reject.txt and all words before it into accept.txt, for example :- 

 ```
90:33   error       Use 'for example' instead of 'e.g.'.
85:22   error       Use 'isn't' instead of 'is not'.
39:14  suggestion  Consider using 'many' instead of 'multiple'.
```
in above three examples, i will add ("for example", "isn't" , "many" ) to accept.js and ( "e.g.", "is not" , "multiple" ) to reject.js is this approach correct ? or i am wrong this time ? please explain 

screen shots of current changes :- 
image

Note:- by mistake i have added node.js to reject.txt file twice, i was attempting to add "Node.js" but by mistake vs code automaticaly adjusted the case after enter and i also did not paid attention at that time so i will correct that in next commit.

File: styles/Vocab/Plone/reject.txt
Comment (stevepiercy) [reply styles/Vocab/Plone/reject.txt:8]: Also all entries must be sorted alphabetically to avoid duplication.

File: styles/Vocab/Plone/reject.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/reject.txt:9]: Before adding these entries, did Vale report them as misspellings?

After adding them, did Vale report that their usage in the docs should be corrected?

if so, then did you correct their usage in the docs?

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by stevepiercy** [comment]
I would have to see the usage to be sure, but in general I agree with the two errors. I think it is safe to make the changes in the docs to silence these errors from Vale.

However, the suggestion depends on context. Keep in mind that Vale is not perfect, and "multiple" might be better than "many". English grammar is needlessly complex. Can you provide more context or its usage?

File: .github/workflows/docs.yml
Comment (chirayu-humar) [reply .github/workflows/docs.yml:41]: sir, i did not changed this manually, but if you want me to revert then i will

Comment (stevepiercy) [reply .github/workflows/docs.yml:41]: Why did you change it, commit it, and push it to this PR?

**Comment by chirayu-humar** [comment]
hello @stevepiercy, 
sorry for tagging you again, i was unsure about, whether you have been notified or not about my last commnent for review, answers to your questions are as follows:- 

before adding these entries, vale reported like this :-
use "Volto" instead of "volto"
use "Plone" instead of "plone"
use "NodeJS" instead of "Nodejs"
use "Razzle" instead of "razzle"

after adding entries, when i ran command "make docs-vale" there were lines in output :-

84:63 error Use 'Plone' instead of 'plone'.
20:38 error Use 'Volto' instead of 'volto'.
127:69 error Avoid using 'node'.
280:19 error Use 'NodeJS' instead of 'Nodejs'.
58:42 error Use 'Razzle' instead of 'razzle'.

so, i we can say that yes; even after adding these terms to reject.txt and accept.txt, errors still persists, Vale still reports that their usage in the docs should be corrected. yes.

comming to third question, i did not correct their usage but my cross question to you sir, do i need to correct their usage in docs along with adding them to reject and accept ? if that is the case then i am ready to do that also, please give instructions to proceed.

**Comment by stevepiercy** [comment]
> so, i we can say that yes; even after adding these terms to reject.txt and accept.txt, errors still persists, Vale still reports that their usage in the docs should be corrected. yes.

You configure Vale with terms in accept.txt and reject.txt, then correct the instances in the docs both _accordingly_ and _depending on their context_. Vale is pretty good about avoiding code and inline literals, but it can make a mistake. Code examples in the docs with `<javascript>` should not be reported as misspellings. When reviewing the recommendations that Vale makes, use your best judgment to update the docs. In any case, changes will be reviewed by a maintainer.

> comming to third question, **i did not correct their usage in docs, but my cross question to you sir, do i need to correct their usage in docs along with adding them to reject and accept ?**

Yes. The whole point of running a spell and grammar checker is to catch misspellings and grammar mistakes for you to correct in the docs. A PR with additions to the configuration and their corresponding corrections in the docs is expected.

For this PR, focus on only a few new terms, then move it out of Draft status for final review.

For Volto only, Vale currently reports:

```console
✖ 822 errors, 378 warnings and 1198 suggestions in 93 files.
```

I would be overjoyed to get 1% of those errors, warnings, and suggestions resolved in one PR.

**Comment by chirayu-humar** [comment]
i have recently made 2 commits into this forked repo in spell-check-update branch, after resolving some errors, majorly i have resolved 2 types of errors i have resolved:- 

- added unidentified words to accept.txt 
- updated the perfect nouns like "Volto", "Plone" etc according to accept.txt and reject.txt 

@stevepiercy, please review the lattest commit, ready to resolve some more errors if every thing is fine till now, waiting for response

**Comment by chirayu-humar** [comment]
i have updated the this pull request (forked repo) with the upstream, because a commit was made into master branch of upstream by @stevepiercy . please review the changes

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:2]: Vale does not warn about acronyms when they are uppercased, such as `API`. `api` is neither a word nor an acronym because of its improper casing, thus we want this term in the `reject.txt` file.

```suggestion
```

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:6]: Move all three above acronyms—css, http, https—into `reject.txt`, per previous comment.

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:8]: Remove only. Acronyms should not cause Vale to emit warnings. Where do you see Vale emit a warning for `LTS`?

```suggestion
```

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:10]: "npm" and "nvm" are good here. They are are the actual and proper names of software. "NPM" and "NVM" would be wrong. These are exceptions to the rule for acronyms.

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:11]: I just checked and the correct term is `Node.js`. It is also a registered trademark, so a variation is also acceptable. See https://nodejs.org/en/about

```suggestion
Node.js
Node.js®
```

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/accept.txt:18]: Remove. "repo" is fine in conversation, but in writing we want to spell out the word as "repository".

```suggestion
```

File: styles/Vocab/Plone/reject.txt
Comment (stevepiercy) [inline styles/Vocab/Plone/reject.txt:7]: 👍

File: styles/Vocab/Plone/reject.txt
Comment (stevepiercy) [reply styles/Vocab/Plone/reject.txt:9]: I'm curious to see the result of Vale's output at each change, showing the counts of warnings, errors, and suggestions, before and after. After corrections are made in the docs, does the configuration change make things worse or improve them?

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:26]: Revert. The correct term is in fact Node.js.

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:48]: Revert. The correct term is in fact Node.js.

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:64]:
```suggestion
another management tool to install/maintain Node.js this step is not needed. If you have less
```

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:127]:
```suggestion
Install the Yarn Classic version (not the 2.x one!), of the popular Node.js package manager.
```

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:209]: Revert. If Vale complains about this term, then add it to `accept.txt`.

Also the original author should have used an inline literal for this, which I think will avoid a Vale warning. Thus:

```suggestion
It also has more information on `plone.volto`.
```

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:269]:
```suggestion
app should be run in a Node.js process (because of the {term}`server-side rendering`
```

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:280]:
```suggestion
2.  Run the Volto Node.js process
```

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:285]:
```suggestion
    to run the Node.js process with the production build. You can also run it manually:
```

File: docs/source/deploying/pm2.md
Comment (stevepiercy) [inline docs/source/deploying/pm2.md:12]:
```suggestion
PM2 is a popular and maintained process manager based in Node.js (https://pm2.keymetrics.io/).
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Progress! I see you have learned some basics, and there are more things to learn, along with some advanced usage. One tip for you, when you use the formal name of a package, I recommend to always check their website for the proper casing and punctuation of the term. Node.js or Node.js® is one of tricky ones, and nvm and npm break the rules of acronyms.

**Review by chirayu-humar — COMMENTED** [review]
I think, I have gone to all the comments now, waiting for further instructions, any changes to make or any further improvement. Till now, i am was only focussing on errors related to perfect nouns, not with suggestions related to vocabs and grammer, am i required to proceed with that because almost all the erros related to perfect nouns are done. waiting for response

**Comment by chirayu-humar** [comment]
sorry, i requested for review before pushing the commit, please review the lattest commit named "changed the left chages". removed "http" and "https" from accept.txt but not included them to reject.txt because i thought it could cause problems.

**Comment by stevepiercy** [comment]
It's difficult for me to follow you. Thus let's focus only on the Vale output. For example:

```console
 docs/source/deploying/apache.md
 12:58   error  Avoid using 'api'.                        Vale.Avoid       
 12:107  error  Punctuation should be inside the quotes.  Microsoft.Quotes 
 12:146  error  Did you really mean 'http'?               Vale.Spelling    
 12:154  error  Did you really mean 'https'?              Vale.Spelling
```

This tells me to look at that file and line to get the context of the usage, and I can make a correct decision. See https://github.com/plone/volto/pull/5056/commits/c4136b0e13cdb668c8a4ad74273ad2b056f000ec to see what I did to fix all 4 of these issues.

```console
docs/source/deploying/pm2.md
 10:22  error       Avoid using 'node'.                 Vale.Avoid             
 12:63  error       Avoid using 'js'.                   Vale.Avoid             
 18:53  error       Did you really mean 'repo'?         Vale.Spelling          
 57:49  error       Use 'that's' instead of 'that is'.  Microsoft.Contractions 
 58:38  suggestion  'ZEO' has no definition.            Microsoft.Acronyms     
```

The first two errors are because Vale considers a period as a boundary character. I tried using a regular expression in `accept.txt`—`Node\.js`—to no effect. However there is a workaround:

```
{term}`Node.js`
```

It's verbose but correct. It is defined in the Glossary, so it will resolve to it and pass spellcheck. Alternatively we could replace all instances of "Node.jd" as an inline literal, but I think that because it is a registered trademark, it should not be an inline literal.

Here's my commit that resolves most of these issues. [6f06daa](https://github.com/plone/volto/pull/5056/commits/6f06daa9b8512099f87066cfb24c45b43afd728d)

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [reply styles/Vocab/Plone/accept.txt]: Vale should not warn about URLs as long as they follow the style rules. See https://github.com/errata-ai/vale/issues/320

If "http" or "https" are used literally as terms and not in URLs, then they should either be inline literals:

```
`http` redirects to `https`
```

...or uppercased to "HTTP" or "HTTPS".

File: styles/Vocab/Plone/accept.txt
Comment (stevepiercy) [reply styles/Vocab/Plone/accept.txt]: OK, let's put LTS in here. I want to make sure that when we add configuration, that we do it for a legitimate reason. Also before adding terms wholesale, we should verify that acronyms follow the pattern recommended by Vale to avoid warnings in the first place. See https://vale.sh/docs/topics/styles/#conditional. In this case, LTS is well known, so I won't fuss over it.

**Comment by stevepiercy** [comment]
I created an issue to get help for that Node.js term thing. It bugs me that we have to use a workaround for it. https://github.com/errata-ai/vale/issues/666

Meanwhile, it is fine to ignore the grammar suggestions.

**Comment by chirayu-humar** [comment]
so shall i convert this pull request from "draft" to "ready for review" ?

**Comment by stevepiercy** [comment]
@chirayu-humar I resolved most of the issues, but not all. When you resolve the remaining issues, specifically with "Node.js", then change it to "ready for review". Thank you!

**Comment by stevepiercy** [comment]
@chirayu-humar I just got a very helpful response from the maintainer of Vale. Would you please see their response, and update according to their guidance: https://github.com/errata-ai/vale/issues/666#issuecomment-1670457942 ? Thank you!

**Comment by chirayu-humar** [comment]
i have merged all the changes made by @stevepiercy but reverted ( `Node.js` to Node.js in line 10 and 12 of "deployment/pm2.md" ) because i could be handled using changes suggested by vale maintainer, and obviously, i have implemented them also please review my lattest commit, if every thing is alright then may i convert the draft into "ready for review" ?

**Review by chirayu-humar — COMMENTED** [review]
reviwed all changes and have merged them to local branch. knowingly left "Javascript" and "javascript" in reject.txt, it could be deleted and still code will work file according to vale maintainer but still left because you also ignored this whild personaly making changes, if need to be changes i can make change and push again.

**Comment by chirayu-humar** [comment]
@stevepiercy, can i convert this pull request from draft to "ready to review", please comment on this so that i can proceed with that, Thanks allot for guiding till now.

**Comment by stevepiercy** [comment]
@chirayu-humar you can do that now. It does mean, after all, "ready to review" not "MERGE ME NOW, RAWR!" 😉

**Comment by stevepiercy** [comment]
@chirayu-humar instead of going through another review cycle, I decided it would be easier to demonstrate what is involved with configuring Vale, especially from the excellent lesson I learned from its maintainer. Please take a look at the configuration and spellings.

I don't think I want to punish a core maintainer by asking them to review the hundreds of docs corrections in this PR, as they are trivial, but at the same time necessary. I'm going to approve this PR as is, and leave it to a core maintainer to merge. @sneridagh @davisagli 

Before:

```
✖ 822 errors, 378 warnings and 1198 suggestions in 93 files.
```

After

```
✖ 602 errors, 380 warnings and 1219 suggestions in 94 files.
```

I opened an [issue for how to handle "plone" and "volto" in MyST labels](https://github.com/errata-ai/vale/issues/666).

Once this PR is merged, then I will update the configuration of Vale in the main Documentation repo.

**Review by stevepiercy — APPROVED** [review]
I approve myself. 😜

**Review by davisagli — APPROVED** [review]

**Pull request merged**
